File: nexcore/src/core/storage/IStorageService.ts
import type { IUploadedFile } from '@forge/shared-types';

/**
 * Storage Service Contract (SOLID Strategy Pattern).
 * Decouples upload handling from specific storage providers (Local Disk, AWS S3, GCS).
 */
export interface IStorageService {
  uploadFile(fileBuffer: Buffer, originalName: string, mimeType: string): Promise<IUploadedFile>;
  deleteFile(filename: string): Promise<boolean>;
  getFileUrl(filename: string): string;
}

File: javacore/src/main/java/com/forge/javacore/core/storage/IStorageService.java
package com.forge.javacore.core.storage;

import org.springframework.web.multipart.MultipartFile;
import com.forge.javacore.core.storage.dto.UploadedFileResponse;

/**
 * Storage Service Interface (SOLID Strategy Pattern).
 * Decouples file uploads from specific storage implementations (Local Disk, AWS S3, GCS).
 */
public interface IStorageService {
    UploadedFileResponse uploadFile(MultipartFile file);
    boolean deleteFile(String filename);
    String getFileUrl(String filename);
}
